__tests__/validations/actions.test.tsx

```diff
@@ -0,0 +1,549 @@
+      {
+        name: 'action_1',
+        ref: {
+          id: 'foo',
+          endpointId: 'bar',
+        },
+        useNewRuntime: true,
+      } as PrefabAction,
+    ],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  validatePrefabs([prefab]);
+
+  t.pass();
+});
+
+test('Throw when component option has a value and a ref object with a value', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [
+      {
+        name: 'action_1',
+        ref: {
+          id: 'foo',
+          endpointId: 'bar',
+        },
+        useNewRuntime: true,
+      } as PrefabAction,
+    ],
+    name: 'Prefab',
+    structure: [
+      {
+        name: 'HelloWorld',
+        options: [
+          {
+            label: 'Action',
+            key: 'actionId',
+            value: '',
+            ref: {
+              value: 'foo',
+            },
+            type: 'ACTION',
+            configuration: {
+              apiVersion: 'v1',
+            },
+          },
+        ],
+        descendants: [],
+      },
+    ],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "structure[0]" failed custom validation because 
+Build error in component HelloWorld: "options[0].value" is not allowed
+ at prefab: Prefab
+`,
+  });
+});
+
+test('Pass when component option has a value and no ref', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    name: 'Prefab',
+    structure: [
+      {
+        name: 'HelloWorld',
+        options: [
+          {
+            label: 'Action',
+            key: 'actionId',
+            value: '',
+            type: 'ACTION',
+            configuration: {
+              apiVersion: 'v1',
+            },
+          },
+        ],
+        descendants: [],
+      },
+    ],
+  } as Prefab;
+
+  validatePrefabs([prefab]);
+
+  t.pass();
+});
+
+test('Pass when component option has a ref and no value', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    name: 'Prefab',
+    structure: [
+      {
+        name: 'HelloWorld',
+        options: [
+          {
+            label: 'Action',
+            key: 'actionId',
+            ref: {
+              value: 'foo',
+            },
+            type: 'ACTION',
+            configuration: {
+              apiVersion: 'v1',
+            },
+          },
+        ],
+        descendants: [],
+      },
+    ],
+  } as Prefab;
+
+  validatePrefabs([prefab]);
+
+  t.pass();
+});
+
+test('Throw when component option has a ref when type is not ACTION', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    name: 'Prefab',
+    structure: [
+      {
+        name: 'HelloWorld',
+        options: [
+          {
+            type: 'TEXT',
+            label: 'Title',
+            key: 'title',
+            ref: {
+              value: 'foo',
```

Comment:
I see `ref.value` here. I think I don't understand the `ref` object? 🤔 Can you explain?

Reply:
The values inside `ref` will be replaced by a unique UUID. The value key will be placed back to the root of the object as follows: `value: 612d76937fbe428cb5cb98e9e1a09aa8`. Keys with the same values inside the ref will get the same UUIDs to connect entities with each other.